### PR TITLE
20 generate a mediadb file for carveme that takes the biolog carbon plates and the carbonless media as input

### PR DESCRIPTION
**Pull request by augeorge**

### CODE REVIEW

**Comment by augeorge** [comment]
PR request overview:

this adds functionality in concerto utils that parses a biolog carbon growth data .csv file, and a minimal media data .csv file and creates a media db pandas dataframe that is consistent with carveme formatting. it includes tests for the functions and testing data

1. added functions: parse_carbon_source_growth_file, parse_carbonless_media_file, create_carveme_mediadb_df to concerto/utils/utils.py
2. added test functions: test_parse_carbon_source_growth_file, test_parse_carbonless_media_file, test_create_carveme_mediadb_df to concerto/testing/test_utils.py
3. added testing data: biolog_carbon_curtobacterium1.csv, CarveMeMinimalMediaFile.csv, csc009C-less_media.csv, expected_carbonless_media_df_from_csc009C_less_media.csv, expected_carveme_mediadb_from_biolog_curtobacterium1_and_csc009C_less_media.csv to concerto/testing/data

all tests are now passing in pytest

File: concerto/testing/create_carveme_mediadb_df.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    """
+    Returns: 
+        df : a dataframe of combined metabolites and
+        carbon sources, per minimal media source
+    """
+    #load carbonless minimal media file
+    dfcmin = pd.read_csv(N-less_media.csv)
+
+    #load biolog growth csv file
+    dfbg = pd.read_csv(biolog_carbon_curtobacterium1.csv)
```
Comment (djinnome) [inline concerto/testing/create_carveme_mediadb_df.py:12]: I think line 12 should be:

```python
dfbg = pd.read_csv(fname)
```

File: concerto/testing/create_carveme_mediadb_df.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import pandas as pd 
+def generate_mediadb(fname) : 
+    """
+    Returns: 
+        df : a dataframe of combined metabolites and
+        carbon sources, per minimal media source
+    """
+    #load carbonless minimal media file
+    dfcmin = pd.read_csv(N-less_media.csv)
```
Comment (djinnome) [inline concerto/testing/create_carveme_mediadb_df.py:9]: This function does not have the right inputs or outputs, and dfcmin should be removed.

File: concerto/testing/create_carveme_mediadb_df.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    #load biolog growth csv file
+    dfbg = pd.read_csv(biolog_carbon_curtobacterium1.csv)
+
+    #import biolog carbon file, assuming from utils.py?
+    metab_growth_names = metab_growth_names
```
Comment (djinnome) [inline concerto/testing/create_carveme_mediadb_df.py:15]: What is going on here?  Where is `metab_growth_names` defined?

File: concerto/testing/create_carveme_mediadb_df.py
Comment (djinnome) [inline concerto/testing/create_carveme_mediadb_df.py:1]: This file should be removed.

**Review by djinnome — CHANGES_REQUESTED** [review]
Looks good. Remove `testing/create_carveme_mediadb_df.py` and I will approve

**Comment by augeorge** [comment]
> Looks good. Remove `testing/create_carveme_mediadb_df.py` and I will approve

@djinnome removed

**Review by djinnome — APPROVED** [review]
looks good to me!

**Pull request merged**